Snowflake.Data/Client/SnowflakeDbConnection.cs

```diff
@@ -37,15 +37,27 @@ public class SnowflakeDbConnection : DbConnection
         // Will fix that in a separated PR though as it's a different issue
         private static Boolean _isArrayBindStageCreated;
 
+        private readonly SnowflakeTomlConnectionBuilder _tomlConnectionBuilder;
+
         protected enum TransactionRollbackStatus
         {
             Undefined, // used to indicate ignored transaction status when pool disabled
             Success,
             Failure
         }
 
-        public SnowflakeDbConnection()
+        public SnowflakeDbConnection() : this(new SnowflakeTomlConnectionBuilder())

```

Comment:
you could use the same singleton instance instead of creating a new one for every connection

Reply:
Done